modal/_container_entrypoint.py

```diff
@@ -394,7 +394,17 @@ def main(container_args: api_pb2.ContainerArguments, client: Client):
     with container_io_manager.heartbeats(is_snapshotting_function), UserCodeEventLoop() as event_loop:
         # If this is a serialized function, fetch the definition from the server
         if function_def.definition_type == api_pb2.Function.DEFINITION_TYPE_SERIALIZED:
-            ser_usr_cls, ser_fun = container_io_manager.get_serialized_function()
+            assert function_def.function_serialized or function_def.class_serialized
+
+            if function_def.function_serialized:
+                ser_fun = deserialize(function_def.function_serialized, _client)
+            else:
+                ser_fun = None
+
+            if function_def.class_serialized:
+                ser_usr_cls = deserialize(function_def.class_serialized, _client)
+            else:
+                ser_usr_cls = None
```

Comment:
I really dislike all these parallel paths for function / class. But I guess we have to live with it.

Reply:
Parts of it we'll have to live with, but as soon as we stop supporting `@build` methods we can remove a bunch of it (the parts that support having *both* a function and a class 😵 )